### ISSUE TITLE
Allow native scroll when element end reached

### ISSUE THREAD

**Issue by JustasKuizinas** [cubiq/iscroll#958]
I have vertical scrollable items in div A. On mobile i need to allow native scroll when i reached scroll start/end and finger is still on div A.How to do that?

**Comment by romainhe** [comment]
I think I had same problem on desktop, here is the fix I wrote on "_wheel" event just before the scrollTo action.

var maxY  = this.maxScrollY;
		var currY = this.y;

		// on scroll down
		if(wheelDeltaY < 0){

			if(maxY === currY){

				// end scroll not prevented
			}else{

				e.preventDefault();
			}
		}
		else if(wheelDeltaY > 0){

			if(currY === 0){

				// end scroll not prevented
			}else{

				e.preventDefault();
			}
		}

**Comment by JustasKuizinas** [comment]
Thanks,

Have you edited source code of the iscroll library?

**Comment by romainhe** [comment]
On github ? Not yet
Le mer. 2 sept. 2015 à 10:29, Justas Kuizinas <notifications@github.com> a
écrit :

**Comment by JustasKuizinas** [comment]
I mean where I have to put your code? Directly into library?

**Comment by romainhe** [comment]
Oh sorry ^^

Yes directly in iscroll.js

At the end of "_wheel" method

I can send you the fixed file if you want

Le mer. 2 sept. 2015 à 11:56, Justas Kuizinas <notifications@github.com> a
écrit :

**Comment by JustasKuizinas** [comment]
Tanks, yes it's working in desktop, for mobile I guess I have to edit _end function. It would be nice if this funcionality would be built into core.

**Comment by romainhe** [comment]
Yep, but I think this plugin is not maintained anymore by cubiq....

**Comment by JustasKuizinas** [comment]
It isnt maintained at all? Because I see that last update was long time ago

**Comment by kaansoral** [comment]
I think he publishes major updates every 2-3 years and let things roll on it's own from that point forward

I recently re-assessed native scrolling, since it's widely available now, however it's not possible to do simple things over the native scroll (webkit overflow scroll for example) - so, iscroll is still a necessary evil

My iscroll is also filled with various improvements, modifications, patches etc.

**Comment by kaansoral** [comment]
`
if(this.options.enable_ofscroll)
		{
			if(wheelDeltaY < 0){
		        if(this.maxScrollY === this.y){

		            // end scroll not prevented
		        }else{

		            e.preventDefault();
					e.stopPropagation();
		        }
		    }
		    else if(wheelDeltaY > 0){

		        if(this.y === 0){

		            // end scroll not prevented
		        }else{

		            e.preventDefault();
					e.stopPropagation();
		        }
		    }
		 }
`

As a coincidence, I also recently needed this feature

I think you neglected to mention that you commented out the initial event routines of the _wheel method, otherwise your method doesn't work

if(!this.options.enable_ofscroll) should be wrapped around the main even cancellation routines of _wheel

This way it can be activated as a feature, I have an on-page scroll element that shouldn't scroll the page, that's why I added this as an option

**Comment by JustasKuizinas** [comment]
And if I want this feature for mobile, I guess I have to edit _end function?
